### PR TITLE
feat: update class events

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

adds events for the class update methods

Closes: #831 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by ryanchristo** [comment]
cc @technicallyty is this still a draft or ready for review?

**Comment by technicallyty** [comment]
> cc @technicallyty is this still a draft or ready for review?

i was gonna wait for https://github.com/regen-network/regen-ledger/pull/863

File: proto/regen/ecocredit/v1/events.proto [hunk 1]
```diff
@@ -130,3 +130,40 @@ message EventCancel {
...
+  repeated string removed_issuers = 3;
+}
+
+// EventClassMetadataUpdated is emitted when the credit class metadata is changed.
+message EventClassMetadataUpdated {
+  // class_name is the name of the class that was updated.
+  string class_name = 1;
+
+  // old_metadata is the metadata before the update.
+  string old_metadata = 2;
```
Comment (technicallyty) [inline proto/regen/ecocredit/v1/events.proto:165]: not sure if its worth to have this field, lmk your opinions!

Comment (ryanchristo) [reply proto/regen/ecocredit/v1/events.proto:165]: I don't have a strong opinion here but including it seems reasonable.

**Review by ryanchristo — APPROVED** [review]
Looks good to me.

**Review by aleem1314 — APPROVED** [review]

**Comment by ryanchristo** [comment]
Can you rerun make proto on this before merging? I should have done so in #879 and we have an error in the generated proto files on master now.

**Pull request merged**